src/main/java/org/dswarm/xsd2jsonschema/JsonSchemaParser.java

```diff
@@ -242,7 +247,7 @@ else if (term instanceof XSModelGroupDefinition) {
 				}).orElseGet(() -> {
 					final JSObject jsElements = new JSObject(elementName, isMixed);
```

Comment:
I looked into that deeper some time (until I figured out that I will be better off with parsing the rng file rather than the xsd file). In my second attempt I did drop the depth counter and installed bookkeeping for the jsobjects, here. The drawback of this approach is that it introduces state. Do you think that's a better way of fixing that?